com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/persistence/DirectLinkingResourceStorageLoadable.java

```diff
@@ -86,12 +89,21 @@ public int readCompressedInt() throws IOException {
     public InternalEObject loadEObject() throws IOException {
       final InternalEObject result = super.loadEObject();
       handleLoadEObject(result, this);
-      objectCount++;
       return result;
     }
 
     @Override
     protected void loadFeatureValue(final InternalEObject internalEObject, final EStructuralFeatureData eStructuralFeatureData) throws IOException {
+      if (recordObjects) {
+        switch (eStructuralFeatureData.kind) {
+        // based on org.eclipse.emf.ecore.resource.impl.BinaryResourceImpl.BinaryIO.FeatureKind.get(EStructuralFeature)
+        // case EOBJECT:
+        // case EOBJECT_CONTAINER_PROXY_RESOLVING:
+        //
+        default:
+          l.add(internalEObject);

```

Comment:
InternalEObject is at the wrong end of the feature. This is why the set you have collected is never big enough.